packages/core/src/sync-realtime/service.ts

```diff
@@ -8,878 +8,633 @@ import {
-  ): Promise<RealtimeLog[]> => {
-    if (!this.hasFactorySource) {
-      return filterLogs({
-        logs,
-        logFilters: this.sources.map((s) => s.criteria),
-      });
-    } else {
+    insertChildAddressLogs,
+    upToBlockNumber,
+  }: {
+    logs: SyncLog[];
+    insertChildAddressLogs: boolean;
+    upToBlockNumber: bigint;
+  },
+) => {
+  if (realtimeSyncService.hasFactorySource === false) {
+    return filterLogs({
+      logs,
+      logFilters: realtimeSyncService.sources.map((s) => s.criteria),
+    });
+  } else {
+    if (insertChildAddressLogs) {
       // Find and insert any new child contracts.
       const matchedFactoryLogs = filterLogs({
         logs,
-        logFilters: this.factorySources.map((fs) => ({
+        logFilters: realtimeSyncService.factorySources.map((fs) => ({
           address: fs.criteria.address,
           topics: [fs.criteria.eventSelector],
         })),
       });
 
-      if (insertChildAddress) {
-        await this.syncStore.insertFactoryChildAddressLogs({
-          chainId: this.network.chainId,
-          logs: matchedFactoryLogs,
-        });
-      }
-
-      // Find any logs matching log filters or child contract filters.
-      // NOTE: It might make sense to just insert all logs rather than introduce
-      // a potentially slow DB operation here. It's a tradeoff between sync
-      // latency and database growth.
-      // NOTE: Also makes sense to hold factoryChildAddresses in memory rather than
-      // a query each interval.
-      const factoryLogFilters = await Promise.all(
-        this.factorySources.map(async (factory) => {
-          const iterator = this.syncStore.getFactoryChildAddresses({
-            chainId: this.network.chainId,
-            factory: factory.criteria,
-            upToBlockNumber: toBlockNumber,
-          });
-          const childContractAddresses: Address[] = [];
-          for await (const batch of iterator) {
-            childContractAddresses.push(...batch);
-          }
-          return {
-            address: childContractAddresses,
-            topics: factory.criteria.topics,
-          };
-        }),
-      );
-
-      return filterLogs({
-        logs,
-        logFilters: [
-          ...this.logFilterSources.map((l) => l.criteria),
-          ...factoryLogFilters,
-        ],
+      await realtimeSyncService.syncStore.insertFactoryChildAddressLogs({
+        chainId: realtimeSyncService.network.chainId,
+        logs: matchedFactoryLogs,
       });
     }
-  };
 
-  /** Returns true if "blocks" has a valid chain of block.parentHash to block.hash. */
-  private isChainConsistent = (
-    blocks: (LightBlock | RealtimeBlock)[],
-  ): boolean => {
-    for (let i = blocks.length - 1; i > 1; i--) {
-      if (blocks[i].parentHash !== blocks[i - 1].hash) return false;
+    // Find any logs matching log filters or child contract filters.
+    // NOTE: It might make sense to just insert all logs rather than introduce
```

Comment:
I'm pretty confident that we could move this filtering operation into SQL. Something like: SELECT log.id FROM VALUES (... pending logs) LEFT JOIN (factory child address clause) ...

Could you create an issue for it? Could be a good one for a beginner.